addons/web/static/src/js/core/utils.js

```diff
@@ -303,6 +303,40 @@ var utils = {
             reader.readAsDataURL(file);
         });
     },
+    /**
+     * Returns an object holding different groups defined by a given criterion
+     * or a default one. Each group is a subset of the original given list.
+     * The given criterion can either be:
+     * - a string: a property name on the list elements which value will be the
+     * group name,
+     * - a function: a handler that will return the group name from a given
+     * element.
+     *
+     * @param {any[]} list
+     * @param {string | (element: any) => string} criterion
+     * @returns {Object}
+     */
+    groupBy: function (list, criterion) {
+        let getter;
+        if (criterion) {
+            switch (typeof criterion) {
+                case 'string': getter = element => element[criterion]; break;
```

Comment:
element need to be an object --> add a check somewhere?

Reply:
Not necessarily; for example if we want to group an array of strings by their length, we might use something like `groupBy(['a', 'aa', 'b'], 'length')` which will return `{ 1: ['a', 'b'], 2: ['aa'] }`. Also since every value in JavaScript is an object it will work with every type (it might yield `undefined` but this is at the developer's discretion to ensure that the property exists).

Reply:
ok